.github/workflows/nightly.yml

```
@@ -122,7 +122,15 @@ jobs:
     needs: check-for-changes
     if: needs.check-for-changes.outputs.should_build == 'true'
-    runs-on: blacksmith-4vcpu-windows-2025
+    # 4 -> 8 vCPU. This job has died on its own timeout twice (runs
+    # 30190411454, and again after PRs #641/#643 added 26 test files), and a
+    # cancelled job never runs its cache-save post step - so each failure left
+    # the next nightly equally cold and it timed out again. Compiling 112
+    # release-mode test binaries is the one workload that genuinely scales
+    # with cores, so doubling them should more than halve wall-clock: ~70min
+    # at $0.016/min = $1.12 becomes ~30min at $0.032/min = $0.96. Cheaper AND
+    # it breaks the timeout death spiral.
+    runs-on: blacksmith-8vcpu-windows-2025
```

actionlint (1.7.12)

[error] label "blacksmith-8vcpu-windows-2025" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)